commit afc88dfda013970bb1e214b331e99adca2f98312
Author: Justin Stitt <justinstitt@google.com>
Date:   Fri Oct 20 19:46:11 2023 +0000

    power: supply: surface-charger: replace deprecated strncpy with strscpy
    
    strncpy() is deprecated for use on NUL-terminated destination strings
    [1] and as such we should prefer more robust and less ambiguous string
    interfaces.
    
    We expect ac->name to be NUL-terminated based on its usage with format
    strings:
    
    surface_charger.c:
    190: ac->psy_desc.name = ac->name;
    
    ...
    
    power_supply_core.c:
    174: dev_dbg(&psy->dev, "%s: Found supply : %s\n",
    175:   psy->desc->name, epsy->desc->name);
    
    Moreover, NUL-padding is not required as ac is already zero-allocated
    before being passed to spwr_ac_init():
    
    surface_charger.c:
    240: ac = devm_kzalloc(&sdev->dev, sizeof(*ac), GFP_KERNEL);
    241: if (!ac)
    242:   return -ENOMEM;
    243:
    244: spwr_ac_init(ac, sdev, p->registry, p->name);
    
    ... this means any future NUL-byte assignments (like the ones that
    strncpy() does) are redundant.
    
    Considering the above, a suitable replacement is `strscpy` [2] due to
    the fact that it guarantees NUL-termination on the destination buffer
    without unnecessarily NUL-padding.
    
    Let's also opt for the more idiomatic strscpy() usage of:
    (dest, src, sizeof(dest))
    
    Link: https://www.kernel.org/doc/html/latest/process/deprecated.html#strncpy-on-nul-terminated-strings [1]
    Link: https://manpages.debian.org/testing/linux-manual-4.8/strscpy.9.en.html [2]
    Link: https://github.com/KSPP/linux/issues/90
    Cc: linux-hardening@vger.kernel.org
    Signed-off-by: Justin Stitt <justinstitt@google.com>
    Link: https://lore.kernel.org/r/20231020-strncpy-drivers-power-supply-surface_charger-c-v1-1-93ddbf668e10@google.com
    Signed-off-by: Sebastian Reichel <sebastian.reichel@collabora.com>

diff --git a/drivers/power/supply/surface_charger.c b/drivers/power/supply/surface_charger.c
index cabdd8da12d0..7a6c62d6f883 100644
--- a/drivers/power/supply/surface_charger.c
+++ b/drivers/power/supply/surface_charger.c
@@ -175,7 +175,7 @@ static void spwr_ac_init(struct spwr_ac_device *ac, struct ssam_device *sdev,
 			 struct ssam_event_registry registry, const char *name)
 {
 	mutex_init(&ac->lock);
-	strncpy(ac->name, name, ARRAY_SIZE(ac->name) - 1);
+	strscpy(ac->name, name, sizeof(ac->name));
 
 	ac->sdev = sdev;